### ISSUE TITLE
NoMethodError in CamaleonCms::Admin#dashboard

### ISSUE THREAD

**Issue by kriskoli** [owen2345/camaleon-cms#687]
I have integrated Camaleon cms with the existing app.But after I have created my admin panel, there is the following issue when try to log in to admin panel.
![screen shot 2017-09-14 at 04 14 48](https://user-images.githubusercontent.com/31746971/30409149-8adcb13e-9903-11e7-9d8a-2f3ebd883ace.png)
Did anyone get the same issue? 
Please help me.

**Comment by owen2345** [comment]
Please put the full error message

**Comment by kriskoli** [comment]
Hi, owen.
I have got the following trouble in terminal.

ActionView::Template::Error (undefined method `name' for {:name=>"Dashboard", :url=>"/en/admin", :options=>{}}:Hash):
    3: <html lang="en">
    4: <head>
    5:     <meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
    6:     <title><%= raw cama_admin_title_draw %></title>
    7:     <meta http-equiv="X-UA-Compatible" content="IE=edge"/>
    8:     <meta name="viewport" content="width=device-width, initial-scale=1"/>
    9:     <%= stylesheet_link_tag "camaleon_cms/admin/admin-manifest", media: "all" %>
  lib/rack/seoredirect.rb:20:in `call'

Thanks.

**Comment by owen2345** [comment]
Fixed! pls test using github version

**Comment by kriskoli** [comment]
Owen, Thanks a lot, 
After github version is installed, the bug is fixed.
But I have another problem now.

ActionView::Template::Error (undefined method `html_safe' for {:tag=>:li, :separator=>""}:Hash):
    1: <% if @breadcrumbs.present? %>
    2:   <ul class="breadcrumb <%= options[:class] %>">
    3:     <% separator = divider.html_safe %>
    4:     <% @breadcrumbs[0..-2].each do |crumb| %>
    5:       <li class="<%= options[:item_class] %>">
    6:         <%= link_to crumb[:name], crumb[:url], crumb[:options] %>
  lib/rack/seoredirect.rb:20:in `call'

Greetings for your help.

**Comment by kriskoli** [comment]
I have fixed that.

Thanks for great CMS.

**Comment by kriskoli** [comment]
Hi, owen.
After I have fixed the problem, there is the following issue.
![screen shot 2017-09-15 at 14 25 22](https://user-images.githubusercontent.com/31746971/30482060-ddfc5ed6-9a21-11e7-8927-dbf7bb6e3480.png)

I have noticed that there is the trouble in breadcrumb_on_rails gem, and fixed the issue by revising 
the <%= render_breadcrumbs :tag => :li, :separator => "" %> to <%= render_breadcrumbs %>.

Regards.

**Comment by owen2345** [comment]
Can you review the logs to check request params and verify if there is slug value?
If it is not present, pls review if there is an javascript error by javascript console which is affecting the slug generator.

**Comment by kriskoli** [comment]
I have reviewed the javascript error in console.
That is :
![screen shot 2017-09-15 at 14 57 37](https://user-images.githubusercontent.com/31746971/30483203-54ec6fa0-9a26-11e7-8cd9-d8f11971c082.png)

![screen shot 2017-09-15 at 14 58 14](https://user-images.githubusercontent.com/31746971/30483232-6d44f450-9a26-11e7-9427-c80adbc9c458.png)

Thanks a lot.

**Comment by brian-kephart** [comment]
I am also experiencing a breadcrumb error since the last commit. Relevant stack trace:

```
FATAL -- : [e941c34e-f0cb-44fe-b1fb-bc91b3a2d0b3] ActionView::Template::Error (undefined method `[]' for #<BreadcrumbsOnRails::Breadcrumbs::Element:0x00007fe0aa1d6f90>):
FATAL -- : [e941c34e-f0cb-44fe-b1fb-bc91b3a2d0b3]     3: <html lang="en">
     4: <head>
     5:     <meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
     6:     <title><%= raw cama_admin_title_draw %></title>
     7:     <meta http-equiv="X-UA-Compatible" content="IE=edge"/>
     8:     <meta name="viewport" content="width=device-width, initial-scale=1"/>
     9:     <%= stylesheet_link_tag "camaleon_cms/admin/admin-manifest", media: "all" %>
FATAL -- : [e941c34e-f0cb-44fe-b1fb-bc91b3a2d0b3]   
FATAL -- : [e941c34e-f0cb-44fe-b1fb-bc91b3a2d0b3] vendor/bundle/ruby/2.4.0/bundler/gems/camaleon-cms-c98ed9f56b15/app/helpers/camaleon_cms/admin/breadcrumb_helper.rb:5:in `block in cama_admin_title_draw'
vendor/bundle/ruby/2.4.0/bundler/gems/camaleon-cms-c98ed9f56b15/app/helpers/camaleon_cms/admin/breadcrumb_helper.rb:5:in `each'
vendor/bundle/ruby/2.4.0/bundler/gems/camaleon-cms-c98ed9f56b15/app/helpers/camaleon_cms/admin/breadcrumb_helper.rb:5:in `cama_admin_title_draw'
vendor/bundle/ruby/2.4.0/bundler/gems/camaleon-cms-c98ed9f56b15/app/views/layouts/camaleon_cms/admin.html.erb:6:in `_vendor_bundle_ruby_______bundler_gems_camaleon_cms_c__ed_f__b___app_views_layouts_camaleon_cms_admin_html_erb___3685030159700069274_70301466949620'

```

Thanks again for all your work on this CMS @owen2345! It really is fantastic.

**Comment by owen2345** [comment]
1: About js error: the problem is jquery version (3x deprecated .size() and other methods), pls use  jquery-migrate.xx
2: apparently there is a conflict with breadcrumb_on_rails gem, can you verify for a other breadcrumb gem?
Regards!!!

**Comment by brian-kephart** [comment]
Perhaps I should have made a separate issue. I commented here because my problem is with the same commit, but it is different than the issue described above. I can't even access the admin portions of my site with the latest version. I see no javascript error because the page doesn't even load.

I'm not using any other breadcrumb gem, just breadcrumbs_on_rails as a dependency of Camaleon.

**Comment by owen2345** [comment]
mmm ok! this weekend I will review with the last version of breadcrumb on rails. Please share me your version of breadcrumb_on_rails gem.

**Comment by brian-kephart** [comment]
I'm using breadcrumbs_on_rails v3.0.1. Currently I've locked Camaleon to commit c5c9019 and it works fine, so it's only the most recent change that caused the issue.

**Comment by brian-kephart** [comment]
I found the bug causing my issue and submitted a PR. I didn't reference this issue because I'm not sure whether it will affect @kriskoli's issue or not.

There was no issue with breadcrumbs_on_rails. It hasn't changed since last year. Ruby just wasn't parsing the new ternary statement as intended, easily [fixed with parentheses](https://github.com/owen2345/camaleon-cms/pull/689/commits/096d2eb8155ad897f239ce62d3859ad4633a02f9).

@kriskoli sorry I hijacked your issue. I thought mine was related, but it appears not.
